go/secretsanta/main.go

```diff
@@ -0,0 +1,66 @@
+package main
+
+import (
+	"log"
+	"math/rand"
+	"time"
+)
+
+var santaList = make([]Pair, 0)
+
+func main() {
+	names := [10]string{"Ally", "Gus",
+		"Roger", "Danil", "Isabel", "Pete",
+		"Rob", "Renee", "Shannon", "Greg"}
+	permutations := make([]Pair, 0)
+	shuffledPermutations := make([]Pair, 0)
+
+	// create all permutations
+	for i := 0; i < len(names); i++ {
+		for j := i + 1; j < len(names); j++ {
+			forwardPair := Pair{santa: names[i], receiver: names[j]}
+			backwardPair := Pair{santa: names[len(names)-(i+1)],
+				receiver: names[len(names)-(j+1)]}
+			permutations = append(permutations, forwardPair, backwardPair)
+		}
+	}
+
+	// randomly shuffle list of permutations
+	rand.Seed(time.Now().UnixNano())
```

Comment:
Don't know enough about go- isn't the default to do this? that's what most languages do.